### PR TITLE
Archiver  and app provider capabilities

### PR DESCRIPTION
**Pull request by wkloucek**
## Description

Adds the archiver and app provider to the ocs capabilities.
Removes unused structs, which already live in REVA

## Related Issue

- Needs https://github.com/cs3org/reva/pull/2088
- Fixes https://github.com/owncloud/ocis/issues/2482

## How Has This Been Tested?

- locally:
```
curl --insecure 'https://localhost:9200/ocs/v1.php/cloud/capabilities?format=json' -u einstein:relativity | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1715  100  1715    0     0   2436      0 --:--:-- --:--:-- --:--:--  2436
{
  "ocs": {
    "meta": {
      "status": "ok",
      "statuscode": 100,
      "message": "OK"
    },
    "data": {
      "capabilities": {
        "core": {
          "pollinterval": 60,
          "webdav-root": "remote.php/webdav",
          "status": {
            "installed": true,
            "maintenance": false,
            "needsDbUpgrade": false,
            "version": "10.0.11.5",
            "versionstring": "10.0.11",
            "edition": "community",
            "productname": "reva"
          },
          "support-url-signing": true
        },
        "checksums": {
          "supportedTypes": [
            "sha1",
            "md5",
            "adler32"
          ],
          "preferredUploadType": "SHA1"
        },
        "files": {
          "privateLinks": false,
          "bigfilechunking": false,
          "undelete": true,
          "versioning": true,
          "favorites": false,
          "blacklisted_files": [],
          "tus_support": {
            "version": "1.0.0",
            "resumable": "1.0.0",
            "extension": "creation,creation-with-upload",
            "max_chunk_size": 0,
            "http_method_override": ""
          },
          "archivers": [
            {
              "enabled": true,
              "version": "2.0",
              "formats": [
                "tar",
                "zip"
              ],
              "archiver_url": "/archiver"
            }
          ],
          "app_providers": [
            {
              "enabled": true,
              "version": "1.0",
              "apps_url": "/app/list",
              "open_url": "/app/open"
            }
          ]
        },
        "dav": {
          "chunking": "",
          "trashbin": "1.0",
          "reports": [
            "search-files"
          ],
          "chunkingParallelUploadDisabled": false
        },
        "files_sharing": {
          "api_enabled": true,
          "resharing": true,
          "group_sharing": true,
          "auto_accept_share": true,
          "share_with_group_members_only": true,
          "share_with_membership_groups_only": true,
          "search_min_length": 3,
          "default_permissions": 22,
          "user_enumeration": {
            "enabled": true,
            "group_members_only": true
          },
          "federation": {
            "outgoing": true,
            "incoming": true
          },
          "public": {
            "enabled": true,
            "send_mail": true,
            "social_share": true,
            "upload": true,
            "multiple": true,
            "supports_upload_only": true,
            "password": {
              "enforced_for": {
                "read_only": true,
                "read_write": true,
                "upload_only": true
              },
              "enforced": true
            },
            "expire_date": {
              "enabled": false
            }
          },
          "user": {
            "send_mail": true,
            "profile_picture": false
          }
        }
      },
      "version": {
        "major": 10,
        "minor": 0,
        "micro": 11,
        "string": "10.0.11",
        "edition": "reva"
      }
    }
  }
}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: storage/pkg/command/frontend.go [hunk 1]
```diff
@@ -54,12 +54,32 @@ func Frontend(cfg *config.Config) *cli.Command {
...
+			archivers := []map[string]interface{}{
+				{
+					"enabled":      true,
+					"version":      "2.0.0",
+					"formats":      []string{"tar", "zip"},
+					"archiver_url": "/archiver",
+				},
+			}
+
+			appproviders := []map[string]interface{}{
```
Comment (pascalwengerter) [inline storage/pkg/command/frontend.go:66]:
```suggestion
			appProviders := []map[string]interface{}{
```

Line 75 suggests we're using camelCase here? If so, please also see line 82 ofc

**Comment by jvillafanez** [comment]
For the archiver, it has a maximum number of files and a maximum size that I think they should be in the capabilities. The client could prevent requests that we know that will fail.

**Comment by wkloucek** [comment]
> For the archiver, it has a maximum number of files and a maximum size that I think they should be in the capabilities. The client could prevent requests that we know that will fail.

Good point :+1: Needs changes in REVA though...

**Comment by kulmann** [comment]
> > For the archiver, it has a maximum number of files and a maximum size that I think they should be in the capabilities. The client could prevent requests that we know that will fail.
> 
> Good point 👍 Needs changes in REVA though...

Yes, good idea for improvement 👍  but please let's get this PR merged first, then improve it (IMO right away ;-) ).

**Review by jvillafanez — APPROVED** [review]
I'm not sure about the changes in the go.mod file, but the rest looks good to me.

File: storage/pkg/flagset/frontend.go [hunk 2]
```diff
@@ -217,6 +217,31 @@ func FrontendWithConfig(cfg *config.Config) []cli.Flag {
...
+			Usage:       "URL where the archiver is reachable",
+			EnvVars:     []string{"STORAGE_FRONTEND_ARCHIVER_URL"},
+			Destination: &cfg.Reva.Archiver.ArchiverURL,
+		},
+
+		// App Provider
+		&cli.StringFlag{
+			Name:        "appprovider-apps-url",
+			Value:       flags.OverrideDefaultString(cfg.Reva.AppProvider.AppsURL, "/app/list"),
+			Usage:       "URL where the archiver is reachable",
```
Comment (kulmann) [inline storage/pkg/flagset/frontend.go:233]:
```suggestion
			Usage:       "URL where the app listing of the app provider is reachable",
```

File: storage/pkg/flagset/frontend.go [hunk 3]
```diff
@@ -217,6 +217,31 @@ func FrontendWithConfig(cfg *config.Config) []cli.Flag {
...
+			EnvVars:     []string{"STORAGE_FRONTEND_APP_PROVIDER_APPS_URL"},
+			Destination: &cfg.Reva.AppProvider.AppsURL,
+		},
+		&cli.StringFlag{
+			Name:        "appprovider-open-url",
+			Value:       flags.OverrideDefaultString(cfg.Reva.AppProvider.OpenURL, "/app/open"),
+			Usage:       "URL where the archiver is reachable",
```
Comment (kulmann) [inline storage/pkg/flagset/frontend.go:240]:
```suggestion
			Usage:       "URL where files can be handed over to an application from the app provider",
```

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**